backend/fleet_management/serializers.py

```diff
@@ -164,3 +165,41 @@ def is_valid(self, raise_exception=False):
             ):
                 err.status_code = status.HTTP_409_CONFLICT
             raise err
+
+
+class RefuelSerializer(serializers.ModelSerializer):
+    driver = UserSerializer(read_only=True)
+    car = CarSerializer()
+
+    class Meta:
+        model = Refuel
+        fields = (
+            "id",
+            "driver",
+            "car",
+            "date",
+            "current_mileage",
+            "refueled_liters",
+            "price_per_liter",
+            "total_cost",
+        )
+
+    def create(self, validated_data):
+        car_id = validated_data.pop("car")["id"]
+        data = self.context["request"].data
+        try:
+            driver = User.objects.get(id=data["driver"]["id"])
+            car = Car.objects.get(id=car_id)
+        except ObjectDoesNotExist as e:
+            raise ValidationError(e.args[0])
+
+        total_cost = Money(
+            currency=data["total_cost.currency"], amount=data["total_cost.amount"]
+        )
+
+        with transaction.atomic():
```

Comment:
why is it wrapped with transaction? you don't seem to create any other object other than Refuel so it seems to me a bit unnecessary.

Reply:
Also I was looking at drive model:

```
        with transaction.atomic():
            drive = Drive.objects.create(
                **validated_data,
                is_verified=is_verified,
                driver=self.context["driver"],
                car=car,
                project=project,
                passenger=passenger
            )
            drive.save()

            return drive
```